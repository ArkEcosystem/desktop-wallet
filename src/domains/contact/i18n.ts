export const translations: { [key: string]: any } = {
	CONTACTS_PAGE: {
		ADD_CONTACT: "Add Contact",
		ADD_CONTACT_MESSAGE: "Add your most frequent contacts for fast, easy payments",
		TITLE: "My Contacts",
		SUBTITLE: "Manage your frequent contacts",
	},
	CONTACT_FORM: {
		ADD_ADDRESS: "Add address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		DELETE_CONTACT: "Delete Contact",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
	MODAL_SEARCH_CONTACT: {
		TITLE: "Search Contact",
		DESCRIPTION: "Find the contact you need quickly and conveniently",
		PLACEHOLDER: "Search...",
	},
	VALIDATION: {
		MAXLENGTH_ERROR: "The Contact Name should have less than {{maxLength}} characters.",
		CONTACT_NAME_EXISTS: "Contact with name {{name}} already exists.",
		CONTACT_NAME_EXISTS_SUFFIX: "already exists",
	},
};
