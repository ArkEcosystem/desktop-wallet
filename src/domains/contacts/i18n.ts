export const translations: { [key: string]: any } = {
	CONTACT_FORM: {
		ADD_ADDRESS: "Add Address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
};
