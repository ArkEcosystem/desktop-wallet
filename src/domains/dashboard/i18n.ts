export const translations: { [key: string]: any } = {
	FILTER_WALLETS: {
		NETWORK: {
			TITLE: "Filter Network",
			DESCRIPTION: "Select the types of networks",
		},
		WALLETS: {
			TITLE: "Your Wallets",
			DESCRIPTION: "Select the display of your wallets",
		},
		CHART: {
			TITLE: "Portfolio Chart",
			DESCRIPTION: "You can hide this unit if you like",
		},
		TRANSACTION_HISTORY: {
			TITLE: "Transaction History",
			DESCRIPTION: "You can hide this unit if you like",
		},
	},

	TRANSACTION_HISTORY: {
		TITLE: "Transactions History",
		EMPTY_TEXT:
			"This will display the history of your transactions. But you don't have more than one transaction at the moment.",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		IMPORT: "Import",
	},

	DASHBOARD_PAGE: {
		CHART: {
			TITLE: "Portfolio Chart",
			PERCENTAGES_LABEL: "Total portfolio",
		},
	},
};
