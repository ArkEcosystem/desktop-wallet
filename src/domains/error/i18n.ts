export const translations: { [key: string]: any } = {
	APPLICATION: {
		TITLE: "Something went wrong",
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		RELOAD: "Reload",
	},
	OFFLINE: {
		TITLE: "Cannot Connect",
		DESCRIPTION: "Check your internet connection if it doesn't work, check it again.",
	},
};
