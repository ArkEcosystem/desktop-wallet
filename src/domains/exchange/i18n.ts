export const translations: { [key: string]: any } = {
	MODAL_ADD_EXCHANGE: {
		TITLE: "Exchange",
		DESCRIPTION:
			"Buy ARK directly from the ARK Desktop Wallet or through one of 20+ exchanges supplying ARK tokens using BTC, ETH or other trading pairs.",
	},
};
