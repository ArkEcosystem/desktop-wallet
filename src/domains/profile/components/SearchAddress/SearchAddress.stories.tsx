import { Wallet } from "@arkecosystem/platform-sdk-profiles";
import { action } from "@storybook/addon-actions";
import { boolean, withKnobs } from "@storybook/addon-knobs";
import React from "react";
import { WalletsDecorator } from "utils/storybook";

import { SearchAddress } from "./SearchAddress";

export default {
	title: "Domains / Profile / Components / SearchAddress",
	decorators: [
		withKnobs,
		(storyFn: any) => (
			<WalletsDecorator count={1} withDelegates={true}>
				{storyFn}
			</WalletsDecorator>
		),
	],
};

export const Default = ({ wallets }: { wallets: Wallet[] }) => (
	<div>
		<SearchAddress
			isOpen={boolean("isOpen", true)}
			wallets={wallets}
			onClose={action("onClose")}
			onAction={console.log}
		/>
	</div>
);
