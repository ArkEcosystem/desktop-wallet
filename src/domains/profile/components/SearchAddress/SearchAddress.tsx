import { Address } from "app/components/Address";
import { Avatar } from "app/components/Avatar";
import { Button } from "app/components/Button";
import { SearchResource } from "app/components/SearchResource";
import { Table } from "app/components/Table";
import React from "react";
import { useTranslation } from "react-i18next";

type SearchContactProps = {
	title?: string;
	description?: string;
	isOpen: boolean;
	wallets: any[];
	onClose?: any;
	onSearch?: any;
	selectActionLabel?: string;
	onAction?: (actionName: string, address: any) => void;
};

const SearchAddressListItem = ({ walletName, address, fiat, balance, onAction, index, selectActionLabel }: any) => (
	<tr className="border-b border-theme-neutral-200">
		<td className="py-6 mt-1">
			<Avatar size="lg" address={address} />
		</td>
		<td className="py-1">
			<Address walletName={walletName} address={address} maxChars={22} />
		</td>
		<td className="font-semibold">
			<div>{balance}</div>
		</td>
		<td className="text-theme-neutral-light">
			<div>{fiat}</div>
		</td>
		<td className="border-b border-dashed border-theme-neutral-200">
			<Button
				data-testid={`AddressListItem__select-${index}`}
				className="float-right"
				variant="plain"
				onClick={() => onAction?.("select" as any, address)}
			>
				{selectActionLabel}
			</Button>
		</td>
	</tr>
);
export const SearchAddress = ({
	isOpen,
	wallets,
	onClose,
	onSearch,
	onAction,
	title,
	description,
	selectActionLabel,
}: SearchContactProps) => {
	const { t } = useTranslation();

	const columns = [
		{
			Header: "Avatar",
			className: "invisible",
		},
		{
			Header: "Name/Address",
		},
		{
			Header: "Balance",
		},
		{
			Header: "Fiat Value",
		},
		{
			Header: "Action",
			className: "invisible",
		},
	];
	return (
		<SearchResource
			isOpen={isOpen}
			title={title || t("PROFILE.MODAL_SEARCH_ADDRESS.TITLE")}
			description={description || t("PROFILE.MODAL_SEARCH_ADDRESS.DESCRIPTION")}
			placeholder="Search..."
			onClose={onClose}
			onSearch={onSearch}
		>
			<Table columns={columns} data={wallets}>
				{({ address, walletName, balance, fiat }: any, index: number) => (
					<SearchAddressListItem
						address={address}
						walletName={walletName}
						balance={balance}
						fiat={fiat}
						index={index}
						selectActionLabel={selectActionLabel}
						onAction={onAction}
					/>
				)}
			</Table>
		</SearchResource>
	);
};

SearchAddress.defaultProps = {
	wallets: [],
	selectActionLabel: "Select",
};
