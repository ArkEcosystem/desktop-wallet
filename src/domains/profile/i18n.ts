export const translations: { [key: string]: any } = {
	MODAL_PROFILE_CREATED: {
		TITLE: "Profile Created!",
		DESCRIPTION_1: "If you are new to the ARK Desktop Wallet, view our tutorial to get started.",
		DESCRIPTION_2: "If not, you can skip the tutorial to go directly to your portfolio.",
		START_TUTORIAL: "Start Tutorial",
		SKIP_TUTORIAL: "Skip Tutorial",
	},
};
