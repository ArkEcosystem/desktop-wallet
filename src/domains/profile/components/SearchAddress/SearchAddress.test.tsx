import { Wallet } from "@arkecosystem/platform-sdk-profiles";
import React from "react";
import { act, env, fireEvent, getDefaultProfileId, render } from "testing-library";

import { translations } from "../../i18n";
import { SearchAddress } from "./SearchAddress";

let wallets: Wallet[];

beforeAll(async () => {
	const profile = env.profiles().findById(getDefaultProfileId());
	await profile.wallets().importByMnemonic("additional wallet", "ARK", "devnet");
	wallets = profile.wallets().values();
});

describe("SearchAddress", () => {
	it("should not render if not open", () => {
		const { asFragment, getByTestId } = render(<SearchAddress isOpen={false} />);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render a modal", () => {
		const { asFragment, getByTestId } = render(<SearchAddress isOpen={true} wallets={wallets} />);

		expect(getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_SEARCH_ADDRESS.TITLE);
		expect(getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_SEARCH_ADDRESS.DESCRIPTION);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render a modal with custom title and description", () => {
		const title = "Modal title";
		const description = "Modal description";
		const { asFragment, getByTestId } = render(
			<SearchAddress isOpen={true} wallets={wallets} title={title} description={description} />,
		);

		expect(getByTestId("modal__inner")).toHaveTextContent("Modal title");
		expect(getByTestId("modal__inner")).toHaveTextContent("Modal description");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should select an address", () => {
		const fn = jest.fn();
		const { getByTestId } = render(<SearchAddress isOpen={true} wallets={wallets} onAction={fn} />);
		act(() => {
			fireEvent.click(getByTestId("AddressListItem__select-0"));
		});
		const selectedWalletAddress = wallets[0]?.address();
		expect(fn).toBeCalledWith("select", selectedWalletAddress);
	});
});
