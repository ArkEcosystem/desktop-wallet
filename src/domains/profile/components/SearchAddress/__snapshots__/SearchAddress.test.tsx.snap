// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAddress should not render if not open 1`] = `<DocumentFragment />`;

exports[`SearchAddress should render a modal 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-50 bg-black opacity-50"
    data-testid="modal__overlay"
  />
  <div
    class="sc-AxhCb hthldP fixed top-0 left-0 right-0 z-50 flex flex-col px-10 pt-6 pb-8 mx-auto mt-24 overflow-hidden rounded-xl bg-theme-background"
    data-testid="modal__inner"
  >
    <div
      class="absolute top-0 right-0 z-50 mt-5 mr-5"
    >
      <button
        class="sc-AxirZ iJKGpA"
        color="neutral"
        data-testid="modal__close-btn"
        type="button"
      >
        <div
          class="p-1"
        >
          <div
            class="sc-AxiKw bopxJn"
            height="12"
            width="12"
          >
            <svg>
              cross-slim.svg
            </svg>
          </div>
        </div>
      </button>
    </div>
    <div
      class="py-4"
    >
      <h2
        class="mb-0 text-3xl font-bold"
      >
        Select address
      </h2>
      <div
        class="flex-1"
      >
        <div
          class="mt-1 text-theme-neutral-dark"
        >
          Find and select an address from your wallets
        </div>
        <div
          class="-mx-12"
        >
          <div
            class="bg-theme-neutral-contrast px-12 pt-8 pb-8 mt-8"
            data-testid="SearchBar"
          >
            <div
              class="flex items-center px-10 py-6 bg-white rounded shadow-xl"
            >
              <div
                class="sc-AxiKw dzBdqj w-4 mr-8 text-theme-neutral-300"
                height="1em"
                width="1em"
              >
                <svg>
                  search.svg
                </svg>
              </div>
              <div
                class="flex-1 pl-4 mr-4 border-l border-theme-neutral-200"
              >
                <input
                  class="sc-AxhUy jqmzti overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 border-none shadow-none"
                  data-testid="Input"
                  placeholder="Search..."
                />
              </div>
              <button
                class="sc-AxirZ dPtiuf my-1"
                color="primary"
                data-testid="SearchBar__button"
                type="button"
              >
                <span
                  class="px-2 text-md"
                >
                  Find it
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="mt-8"
        >
          <div
            class="sc-fzplWN tcxiV"
            role="table"
          >
            <table
              class="table-auto"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--0"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top invisible"
                    >
                      <div>
                        Avatar
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--1"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                        Name/Address
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--2"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                        Balance
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--3"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                        Fiat Value
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--4"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top invisible"
                    >
                      <div>
                        Action
                      </div>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  class="border-b border-theme-neutral-200"
                >
                  <td
                    class="py-6 mt-1"
                  >
                    <div
                      class="Avatar__AvatarWrapper-sc-1vsy2s2-0 ikUhMz"
                      data-testid="Avatar"
                    >
                      <div
                        class="w-full h-full"
                      >
                        <img
                          alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                          src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                          title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                        />
                      </div>
                    </div>
                  </td>
                  <td
                    class="py-1"
                  >
                    <div
                      class="inline-block truncate align-middle"
                    >
                      <span
                        class="text-theme-text max-w-24 flex-auto truncate mt-4 mr-1 font-semibold text-base"
                        data-testid="address__wallet-name"
                      >
                        ARK Wallet 1
                      </span>
                      <span
                        class="sc-AxjAm StDqN text-theme-neutral-light font-semibold text-base"
                        data-testid="address__wallet-address"
                      >
                        D8rr7B1d6â€¦VBMs6YUYD
                      </span>
                    </div>
                  </td>
                  <td
                    class="font-semibold"
                  >
                    <div>
                      120 ARK
                    </div>
                  </td>
                  <td
                    class="text-theme-neutral-light"
                  >
                    <div>
                      980 USD
                    </div>
                  </td>
                  <td
                    class="border-b border-dashed border-theme-neutral-200"
                  >
                    <button
                      class="sc-AxirZ fhzkRl float-right"
                      color="primary"
                      data-testid="AddressListItem__select-0"
                      type="button"
                    >
                      Select
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b border-theme-neutral-200"
                >
                  <td
                    class="py-6 mt-1"
                  >
                    <div
                      class="Avatar__AvatarWrapper-sc-1vsy2s2-0 ikUhMz"
                      data-testid="Avatar"
                    >
                      <div
                        class="w-full h-full"
                      >
                        <img
                          alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                          src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                          title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                        />
                      </div>
                    </div>
                  </td>
                  <td
                    class="py-1"
                  >
                    <div
                      class="inline-block truncate align-middle"
                    >
                      <span
                        class="text-theme-text max-w-24 flex-auto truncate mt-4 mr-1 font-semibold text-base"
                        data-testid="address__wallet-name"
                      >
                        ARK Wallet 2
                      </span>
                      <span
                        class="sc-AxjAm StDqN text-theme-neutral-light font-semibold text-base"
                        data-testid="address__wallet-address"
                      >
                        D8rr7B1d6â€¦VBMs6YUYD
                      </span>
                    </div>
                  </td>
                  <td
                    class="font-semibold"
                  >
                    <div>
                      120 ARK
                    </div>
                  </td>
                  <td
                    class="text-theme-neutral-light"
                  >
                    <div>
                      980 USD
                    </div>
                  </td>
                  <td
                    class="border-b border-dashed border-theme-neutral-200"
                  >
                    <button
                      class="sc-AxirZ fhzkRl float-right"
                      color="primary"
                      data-testid="AddressListItem__select-1"
                      type="button"
                    >
                      Select
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SearchAddress should render a modal with custom title and description 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-50 bg-black opacity-50"
    data-testid="modal__overlay"
  />
  <div
    class="sc-AxhCb hthldP fixed top-0 left-0 right-0 z-50 flex flex-col px-10 pt-6 pb-8 mx-auto mt-24 overflow-hidden rounded-xl bg-theme-background"
    data-testid="modal__inner"
  >
    <div
      class="absolute top-0 right-0 z-50 mt-5 mr-5"
    >
      <button
        class="sc-AxirZ iJKGpA"
        color="neutral"
        data-testid="modal__close-btn"
        type="button"
      >
        <div
          class="p-1"
        >
          <div
            class="sc-AxiKw bopxJn"
            height="12"
            width="12"
          >
            <svg>
              cross-slim.svg
            </svg>
          </div>
        </div>
      </button>
    </div>
    <div
      class="py-4"
    >
      <h2
        class="mb-0 text-3xl font-bold"
      >
        Modal title
      </h2>
      <div
        class="flex-1"
      >
        <div
          class="mt-1 text-theme-neutral-dark"
        >
          Modal description
        </div>
        <div
          class="-mx-12"
        >
          <div
            class="bg-theme-neutral-contrast px-12 pt-8 pb-8 mt-8"
            data-testid="SearchBar"
          >
            <div
              class="flex items-center px-10 py-6 bg-white rounded shadow-xl"
            >
              <div
                class="sc-AxiKw dzBdqj w-4 mr-8 text-theme-neutral-300"
                height="1em"
                width="1em"
              >
                <svg>
                  search.svg
                </svg>
              </div>
              <div
                class="flex-1 pl-4 mr-4 border-l border-theme-neutral-200"
              >
                <input
                  class="sc-AxhUy jqmzti overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 border-none shadow-none"
                  data-testid="Input"
                  placeholder="Search..."
                />
              </div>
              <button
                class="sc-AxirZ dPtiuf my-1"
                color="primary"
                data-testid="SearchBar__button"
                type="button"
              >
                <span
                  class="px-2 text-md"
                >
                  Find it
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="mt-8"
        >
          <div
            class="sc-fzplWN tcxiV"
            role="table"
          >
            <table
              class="table-auto"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--0"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top invisible"
                    >
                      <div>
                        Avatar
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--1"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                        Name/Address
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--2"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                        Balance
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--3"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                        Fiat Value
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--4"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top invisible"
                    >
                      <div>
                        Action
                      </div>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  class="border-b border-theme-neutral-200"
                >
                  <td
                    class="py-6 mt-1"
                  >
                    <div
                      class="Avatar__AvatarWrapper-sc-1vsy2s2-0 ikUhMz"
                      data-testid="Avatar"
                    >
                      <div
                        class="w-full h-full"
                      >
                        <img
                          alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                          src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                          title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                        />
                      </div>
                    </div>
                  </td>
                  <td
                    class="py-1"
                  >
                    <div
                      class="inline-block truncate align-middle"
                    >
                      <span
                        class="text-theme-text max-w-24 flex-auto truncate mt-4 mr-1 font-semibold text-base"
                        data-testid="address__wallet-name"
                      >
                        ARK Wallet 1
                      </span>
                      <span
                        class="sc-AxjAm StDqN text-theme-neutral-light font-semibold text-base"
                        data-testid="address__wallet-address"
                      >
                        D8rr7B1d6â€¦VBMs6YUYD
                      </span>
                    </div>
                  </td>
                  <td
                    class="font-semibold"
                  >
                    <div>
                      120 ARK
                    </div>
                  </td>
                  <td
                    class="text-theme-neutral-light"
                  >
                    <div>
                      980 USD
                    </div>
                  </td>
                  <td
                    class="border-b border-dashed border-theme-neutral-200"
                  >
                    <button
                      class="sc-AxirZ fhzkRl float-right"
                      color="primary"
                      data-testid="AddressListItem__select-0"
                      type="button"
                    >
                      Select
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b border-theme-neutral-200"
                >
                  <td
                    class="py-6 mt-1"
                  >
                    <div
                      class="Avatar__AvatarWrapper-sc-1vsy2s2-0 ikUhMz"
                      data-testid="Avatar"
                    >
                      <div
                        class="w-full h-full"
                      >
                        <img
                          alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                          src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                          title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                        />
                      </div>
                    </div>
                  </td>
                  <td
                    class="py-1"
                  >
                    <div
                      class="inline-block truncate align-middle"
                    >
                      <span
                        class="text-theme-text max-w-24 flex-auto truncate mt-4 mr-1 font-semibold text-base"
                        data-testid="address__wallet-name"
                      >
                        ARK Wallet 2
                      </span>
                      <span
                        class="sc-AxjAm StDqN text-theme-neutral-light font-semibold text-base"
                        data-testid="address__wallet-address"
                      >
                        D8rr7B1d6â€¦VBMs6YUYD
                      </span>
                    </div>
                  </td>
                  <td
                    class="font-semibold"
                  >
                    <div>
                      120 ARK
                    </div>
                  </td>
                  <td
                    class="text-theme-neutral-light"
                  >
                    <div>
                      980 USD
                    </div>
                  </td>
                  <td
                    class="border-b border-dashed border-theme-neutral-200"
                  >
                    <button
                      class="sc-AxirZ fhzkRl float-right"
                      color="primary"
                      data-testid="AddressListItem__select-1"
                      type="button"
                    >
                      Select
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
